astropy/coordinates/sky_coordinate.py

```diff
@@ -218,49 +208,69 @@ class or the corresponding string alias.  The frame classes that are built in
 
     def __init__(self, *args, copy=True, **kwargs):
 
-        # Parse the args and kwargs to assemble a sanitized and validated
-        # kwargs dict for initializing attributes for this object and for
-        # creating the internal self._sky_coord_frame object
-        args = list(args)  # Make it mutable
-        kwargs = self._parse_inputs(args, kwargs)
-
-        frame = kwargs['frame']
-        frame_attr_names = frame.get_frame_attr_names()
-
         # these are frame attributes set on this SkyCoord but *not* a part of
         # the frame object this SkyCoord contains
         self._extra_frameattr_names = set()
 
-        for attr in kwargs:
-            if (attr not in frame_attr_names and
-                attr in frame_transform_graph.frame_attributes):
-                # Setting it will also validate it.
-                setattr(self, attr, kwargs[attr])
+        # If all that is passed in is a frame instance that already has data,
+        # we should bypass all of the parsing and logic below. This is here
+        # to make this the fastest way to create a SkyCoord instance. Many of
+        # the classmethods implemented for performance enhancements will use
+        # this in the initialization path
```

Comment:
Extremely minor nitpick:
```suggestion
        # this as the initialization path
```